### ISSUE TITLE
Bad comparison of unsigned vs zero in update_lmb_associativity_index()

### ISSUE THREAD

**Issue by mpe** [linuxppc/linux#189]
Coverity spotted:

```
*** CID 188226:  Control flow issues  (NO_EFFECT)
/arch/powerpc/platforms/pseries/hotplug-memory.c: 207 in update_lmb_associativity_index()
201     	}
202     
203     	aa_index = find_aa_index(dr_node, ala_prop, lmb_assoc);
204     
205     	dlpar_free_cc_nodes(lmb_node);
206     
>>>     CID 188226:  Control flow issues  (NO_EFFECT)
>>>     This less-than-zero comparison of an unsigned value is never true. "aa_index < 0U".
207     	if (aa_index < 0) {
208     		pr_err("Could not find LMB associativity\n");
209     		return -1;
210     	}
211     
212     	lmb->aa_index = aa_index;
```

Looking at `find_aa_index()` it actually returns -1 in several places, so that probably needs fixing also.

**Comment by leitao** [comment]
This seems to be fixed already by:

```commit b45e9d761ba2d60044b610297e3ef9f947ac157f
Author: YueHaibing <yuehaibing@huawei.com>
Date:   Tue Oct 9 21:59:13 2018 +0800

    powerpc/pseries/memory-hotplug: Fix return value type of find_aa_index
    
    The variable 'aa_index' is defined as an unsigned value in
    update_lmb_associativity_index(), but find_aa_index() may return -1
    when dlpar_clone_property() fails. So change find_aa_index() to return
    a bool, which indicates whether 'aa_index' was found or not.
    
    Fixes: c05a5a40969e ("powerpc/pseries: Dynamic add entires to associativity lookup array")
    Signed-off-by: YueHaibing <yuehaibing@huawei.com>
```

**Comment by mpe** [comment]
Yes, fixed in b45e9d761ba2. Thanks YueHaibing :cake:
